docs/src/proposals/optimistic_confirmation.md

```diff
@@ -0,0 +1,237 @@
+# Optimistic Confirmation
+
+## Primitives
+
+`vote(X, S)` - Votes will be augmented with a "reference" slot, `X` 
+which is the **last** ancestor at which this validator voted on this fork
+with a proof of switching. Thus as long as a validator keeps voting on the 
+same fork, `X` will remain unchanged.All votes will then be of the form
+`vote(X, S)`, where `S` is the sorted list of slots `(s, s.lockout)`
+being voted for.
+
+Given a vote `vote(X, S)`, let `S.last == vote.last` be the last slot in `S`.
+
+* `Claim`: if a validator submits `vote(X, S)`, the same validator
+should not have voted on a slot `s'` where  `X < s' < S.last` and `s'`
+is not a descendant of `X` and an ancestor of `S.last`. 
+
+* `Proof`: First note that the vote for `S` must have come after the vote
+for `S'` (due to lockouts, the higher vote must have after). Thus, the sequence 
+of votes must have been: `X ... S' ... S`. This means after the second vote the
+validator must have switched back to the original fork at some `X' > X`. This 
+means the vote for `S` should have used `X'` as the "reference" point, so the 
+vote should have been of the form `vote(X', S)`, not `vote(X, S)`
+(Recall the vote should contain the `latest` switch).
+
+To enforce this, we define the "Optimistic Slashing" slashing conditions. Given
+any two distinct votes `vote(X, S)`and `vote(X', S')` by the same validator,
+the votes must satisfy (namely the last two rules imply the ranges cannot
+overlap):
+
+* `X <= S.last`, `X' <= S'.last`
+* All `s` in `S.last` are ancestors/descendants of one another, 
+all `s'` in `S'.last` areancsestors/desccendants of one another,
```

Comment:
`areaancestors` needs a space

Reply:
X must be an ancestor of a slot in S?

Reply:
Of every slot in S, it's the slot at which this validator last "switched" to this fork